File: go/guard_model.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// GuardModelService — V3 guard model training & performance.
type GuardModelService struct {
	t *internal.Transport
}

func (s *GuardModelService) Stats(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v3/guard-model/stats", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *GuardModelService) Performance(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v3/guard-model/performance", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *GuardModelService) Train(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/guard-model/train", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *GuardModelService) TrainStatus(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v3/guard-model/train/status", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *GuardModelService) TrainCancel(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/guard-model/train/cancel", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

File: rust/src/resources/guard_model.rs
use std::sync::Arc;

use reqwest::Method;
use serde_json::Value;

use crate::error::Result;
use crate::transport::Transport;

pub struct GuardModel {
    transport: Arc<Transport>,
}

impl GuardModel {
    pub(crate) fn new(transport: Arc<Transport>) -> Self {
        Self { transport }
    }

    pub async fn stats(&self) -> Result<Value> {
        self.transport
            .request(Method::GET, "/v3/guard-model/stats", None::<&()>)
            .await
    }

    pub async fn performance(&self) -> Result<Value> {
        self.transport
            .request(Method::GET, "/v3/guard-model/performance", None::<&()>)
            .await
    }

    pub async fn train(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/guard-model/train", Some(body))
            .await
    }

    pub async fn train_status(&self) -> Result<Value> {
        self.transport
            .request(Method::GET, "/v3/guard-model/train/status", None::<&()>)
            .await
    }

    pub async fn train_cancel(&self) -> Result<Value> {
        self.transport
            .request(
                Method::POST,
                "/v3/guard-model/train/cancel",
                None::<&()>,
            )
            .await
    }
}
